packages/lambda-powertools-pattern-basic/__tests__/supplement-csv.js

```diff
@@ -0,0 +1,71 @@
+const supplementCsv = require('../supplement-csv')
+
+describe('supplement csv', () => {
+  describe('when there are no existing items defined', () => {
+    it('should return blank string when nothing to add', () => {
+      const expected = ''
+
+      const actual = supplementCsv()
+
+      expect(actual).toEqual(expected)
+    })
+
+    it('should return csv string of key value pairs provided', () => {
+      const expected = 'key1:value1,key2:value2,key3:value3'
+      const inputs = {
+        additional: 'key1:value1,key2:value2,key3:value3'
+      }
+
+      const actual = supplementCsv(inputs)
+
+      expect(actual).toEqual(expected)
+    })
+  })
+
+  describe('when there are no items to add', () => {
+    it('should return existing csv string', () => {
+      const expected = 'key1:value1,key2:value2,key3:value3'
+      const inputs = {
+        existing: 'key1:value1,key2:value2,key3:value3'
+      }
+
+      const actual = supplementCsv(inputs)
+
+      expect(actual).toEqual(expected)
+    })
+  })
+
+  describe('when there is an existing csv string', () => {
+    it('should supplement with items to add', () => {
+      const expected = 'key1:value1,key2:value2,key3:value3,key4:value4,key5:value5,key6:value6'.split(',')
+      const inputs = {
+        existing: 'key1:value1,key2:value2,key3:value3',
+        additional: 'key4:value4,key5:value5,key6:value6'
+      }
+
+      const actual = supplementCsv(inputs).split(',')
+
+      actual.forEach(pair => {
+        expect(expected).toContainEqual(pair)
+      })
+
+      expect(actual.length).toEqual(expected.length)
+    })
+
+    it('should not overwrite existing keys with items to add', () => {
+      const expected = 'key1:value1,key2:value2,key3:value3,key5:value5,key6:value6'.split(',')
+      const inputs = {
+        existing: 'key1:value1,key2:value2,key3:value3',
+        additional: 'key1:newValue1,key5:value5,key6:value6'
+      }
+
+      const actual = supplementCsv(inputs).split(',')
+
+      actual.forEach(pair => {
+        expect(expected).toContainEqual(pair)
+      })
+
+      expect(actual.length).toEqual(expected.length)
+    })
+  })
```

Comment:
Need another test case to make sure we handle inputs like `env:stage:test` (see below), and when the tag is just a `value` not `key:value`

Reply:
datadog supports both `value` and `key:value` for tags

![image](https://user-images.githubusercontent.com/546969/60057027-01942100-96db-11e9-8cb3-b03a0021c44d.png)

Reply:
That makes things interesting. Very good spot!